packages/lib/src/vis/shared/SelectionTool.tsx

```diff
@@ -77,24 +82,49 @@ function SelectionTool(props: Props) {
       const { sourceEvent, unprojectedPoint } = evt;
       const { target, pointerId } = sourceEvent;
       (target as Element).releasePointerCapture(pointerId);
+      setStartPoint(undefined);
+      setEndPoint(undefined);
 
-      const point = worldToData(boundPointToFOV(unprojectedPoint, camera));
+      if (!modifierKeyPressed) {
+        return;
+      }
 
       if (onSelectionEnd) {
         onSelectionEnd({
           startPoint,
-          endPoint: point,
+          endPoint: worldToData(boundPointToFOV(unprojectedPoint, camera)),
         });
       }
-      setStartPoint(undefined);
-      setEndPoint(undefined);
     },
-    [startPoint, camera, worldToData, onSelectionEnd]
+    [startPoint, modifierKeyPressed, worldToData, camera, onSelectionEnd]
+  );
+
+  const onKeyDown = useCallback(
+    (evt: KeyboardEvent) => {
+      setModifierPressed(checkModifierKey(modifierKey, evt));
+    },
+    [modifierKey]
+  );
+
+  const onKeyUp = useCallback(
+    (evt: KeyboardEvent) =>
+      setModifierPressed(checkModifierKey(modifierKey, evt)),
+    [modifierKey]
   );
 
   useCanvasEvents({ onPointerDown, onPointerMove, onPointerUp });
 
-  if (!startPoint || !endPoint) {
+  useEffect(() => {
+    document.addEventListener('keydown', onKeyDown);
+    document.addEventListener('keyup', onKeyUp);
+
+    return () => {
+      document.removeEventListener('keydown', onKeyDown);
+      document.removeEventListener('keyup', onKeyUp);
+    };
+  });
```

Comment:
I attached key events on `document` rather than on the `canvas` because the `canvas` is not focusable and therefore does not trigger key events.

Reply:
`window` would be an even better choice. That being said, I just remembered that react-hookz provides a hook for this! https://react-hookz.github.io/web/?path=/docs/dom-usekeyboardevent--example

Can't believe I didn't think of it, but you could also use https://react-hookz.github.io/web/?path=/docs/dom-useeventlistener--example in `useCanvasEvents`!

Reply:
Second part of my comment done in #1007